core/src/components/datetime/datetime.tsx

```diff
@@ -2001,6 +2029,37 @@ export class Datetime implements ComponentInterface {
               }
             }
 
+            let dateStyle: DatetimeHighlightStyle | undefined = undefined;
+
+            /**
+             * Custom highlight styles should not override the style for selected dates,
+             * nor apply to "filler days" at the start of the grid.
+             */
+            if (highlightedDates !== undefined && !isActive && day !== null) {
+              if (Array.isArray(highlightedDates)) {
+                const matchingHighlight = highlightedDates.find((hd) => hd.date === dateIsoString);
```

Comment:
We might want to consider removing time values from the ISO 8601 string. For example, this is a valid input for the `date` key: `2023-02-01T16:30`. However, we do a strict equality with the ISO 8601 string for just the date, so the example date I passed would never match anything.